finagle-http/src/main/scala/com/twitter/finagle/http/Codec.scala

```diff
@@ -521,3 +522,108 @@ private[finagle] class HttpServerTracingFilter[Req <: Request, Res](serviceName:
     service(request)
   }
 }
+
+private[finagle] object OpenTracing {
+
+  import io.opentracing.Tracer
+  import io.opentracing.Span
+  import java.util.HashMap
+  import scala.collection.JavaConverters._
+
+  /**
+   * @param span the parent span of all requests created during the
+   *  scope of the enclosed function.
+   * @param rpcName the operation name for this RPC 
+   */
+  def letSpanAndRpc[R](span: Span, rpcName: String)(f: => R): R = {
```

Comment:
let's do something like `letParentSpanAndOperationName` or `letSpanAndOperationName`???? kind of a bummer to do these in one line since they do not refer to the same span.